processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -432,9 +453,9 @@ private void applyPropertyNameBasedMapping() {
                             newPropertyMapping = new PropertyMappingBuilder()
                                 .mappingContext( ctx )
                                 .sourceMethod( method )
+                                .targetPropertyName( targetPropertyName )
```

Comment:
Some of the builder code introduced a set order dependency... Not sure if that is ok